src/repository/postgres/doctor.js

```diff
@@ -0,0 +1,83 @@
+export default class PGDoctor {
+  constructor(pool) {
+    this.pool = pool;
+  }
+
+  getAll = async () => {
+    const result = [];
+
+    const query = `
+      SELECT doctors.*, specializations.title 
+      FROM doctors INNER JOIN specializations 
+        ON doctors.specialization_id = specializations.id
+    `;
+
+    const doctors = await this.pool.query(query);
+
+    doctors.rows.forEach((doctor) => {
+      result.push({
+        id: doctor.id,
+        userId: doctor.user_id,
+        name: doctor.name,
+        specialization: doctor.title,
+      });
+    });
+
+    return result;
+  };
+
+  getById = async (id) => {
+    const query = `
+      SELECT doctors.*, specializations.title 
+      FROM doctors INNER JOIN specializations 
+        ON doctors.specialization_id = specializations.id 
+      WHERE doctors.id = $1
+    `;
+
+    const doctor = await this.pool.query(query, [id]);
+
+    if (!doctor.rows.length) {
+      return null;
+    }
+
+    const {
+      rows: [doctorInfo],
+    } = doctor;
+
+    const result = {
```

Comment:
Looks like this is the same object map. Could be moved to `mapper` method and reuse for different repository methods.

Reply:
:white_check_mark: Done